File: front/desafiocertponto/src/models/apiResponse.ts
export class ApiResponse<T> {
  success: boolean;
  data: T | null | undefined;
  errorMessage: string;

  constructor(
    success: boolean,
    data: T | null | undefined,
    errorMessage: string = ""
  ) {
    this.success = success;
    this.data = data;
    this.errorMessage = errorMessage;
  }

  static successResponse<T>(data: T): ApiResponse<T> {
    return new ApiResponse<T>(true, data);
  }

  static errorResponse<T>(errorMessage: string): ApiResponse<T> {
    return new ApiResponse<T>(false, null, errorMessage);
  }
}

File: api/DesafioCertponto/DesafioCertponto.Domain/Model/ApiResponse.cs
﻿namespace DesafioCertponto.Domain.Model
{
    public class ApiResponse<T>
    {
        public bool Success { get; set; }
        public T Data { get; set; }
        public string ErrorMessage { get; set; }

        public ApiResponse(bool success, T data, string errorMessage = null)
        {
            Success = success;
            Data = data;
            ErrorMessage = errorMessage;
        }

        public static ApiResponse<T> SuccessResponse(T data)
        {
            return new ApiResponse<T>(true, data);
        }

        public static ApiResponse<T> ErrorResponse(string errorMessage)
        {
            return new ApiResponse<T>(false, default, errorMessage);
        }
    }
}
